lms/static/js/student_account/views/account_settings_factory.js

```diff
@@ -246,6 +247,10 @@
                 }
             ];
 
+            if (!collectYearOfBirth){
+              aboutSectionsData[1]['fields'].splice(2,1)
```

Comment:
Can we get the index of the YOB field to splice? Assuming if someone changes the order of the fields list it will remove an incorrect field. You can get the idea from email field here https://github.com/edx/edx-platform/blob/a48ab68c9f02646b744afc9c8694f8af180b5b68/lms/static/js/student_account/views/account_settings_factory.js#L269-L273